File: app/src/main/java/com/hjc/base/viewmodel/LiveViewModel.java
package com.hjc.base.viewmodel;


import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;

import com.hjc.base.bean.UserBean;
import com.hjc.base.model.LiveModel;
import com.hjc.baselib.viewmodel.BaseViewModel;


public class LiveViewModel extends BaseViewModel {

    private final LiveModel liveModel;

    private final MutableLiveData<UserBean> userLiveData = new MutableLiveData<>();

    public LiveViewModel(@NonNull Application application) {
        super(application);
        liveModel = new LiveModel();
    }

    public void showData() {
        UserBean bean = liveModel.getUser();
        userLiveData.setValue(bean);
    }

    public void changeData() {
        UserBean bean = new UserBean("李四", 28, true);
        userLiveData.setValue(bean);
    }

    public void hideData() {
        UserBean bean = liveModel.getUser();
        bean.setShow(false);
        userLiveData.setValue(bean);
    }

    public MutableLiveData<UserBean> getUserLiveData() {
        return userLiveData;
    }
}

File: module-home/src/main/java/com/hjc/module_home/viewmodel/live/LiveViewModel.kt
package com.hjc.module_home.viewmodel.live

import android.app.Application
import androidx.lifecycle.MutableLiveData
import com.hjc.library_base.viewmodel.BaseViewModel
import com.hjc.module_home.entity.UserBean
import com.hjc.module_home.model.LiveModel

class LiveViewModel(application: Application) : BaseViewModel(application) {

    private val liveModel: LiveModel = LiveModel()

    val userLiveData: MutableLiveData<UserBean> = MutableLiveData<UserBean>()

    fun showData() {
        val bean: UserBean = liveModel.user
        userLiveData.value = bean
    }

    fun changeData() {
        val bean = UserBean("李四", 28, true)
        userLiveData.value = bean
    }

    fun hideData() {
        val bean: UserBean = liveModel.user
        bean.isShow = false
        userLiveData.value = bean
    }

}